HPXMLtoOpenStudio/resources/hvac.rb

```diff
@@ -3555,6 +3570,135 @@ def self.apply_installation_quality(model, heating_system, cooling_system, unita
     program_calling_manager.addProgram(fault_program)
   end
 
+  def self.calculate_heat_pump_defrost_load_power_watts(heat_pump, unit_multiplier, design_airflow, max_heating_airflow, fan_watts_per_cfm)
+    # Calculate q_dot and p_dot
+    # q_dot is used for EMS program to account for introduced cooling load and supp coil power consumption by actuating other equipment objects
+    # p_dot is used for calculating coil defrost compressor power consumption
+    return [nil, nil] unless heat_pump.advanced_defrost_approach
+
+    is_ducted = !heat_pump.distribution_system_idref.nil?
+    # determine defrost cooling rate and defrost cooling cop based on whether ducted
+    if is_ducted
+      # 0.45 is from Jon's lab and field data analysis, defrost is too short to reach steady state so using cutler curve is not correct
+      # 1.0 is from Jon's lab and field data analysis, defrost is too short to reach steady state so using cutler curve is not correct
+      # Transient effect already accounted
+      capacity_defrost_multiplier = 0.45
+      cop_defrost_multiplier = 1.0
+    else
+      capacity_defrost_multiplier = 0.1
+      cop_defrost_multiplier = 0.08
+    end
+    nominal_cooling_capacity_1x = heat_pump.cooling_capacity / unit_multiplier
+    max_heating_airflow_1x = max_heating_airflow / unit_multiplier
+    design_airflow_1x = design_airflow / unit_multiplier
+    defrost_flow_fraction = max_heating_airflow_1x / design_airflow_1x
+    defrost_power_fraction = defrost_flow_fraction**3
+    power_design = fan_watts_per_cfm * design_airflow_1x
+    p_dot_blower = power_design * defrost_power_fraction
+    p_dot_odu_fan = 44.348 * UnitConversions.convert(nominal_cooling_capacity_1x, 'Btu/hr', 'ton') + 62.452
```

Comment:
Maybe add a comment on this regression. It's based on manufacturer data for ~70 systems ranging from 1.5 to 5 tons with varying efficiency levels.